model/lifecycle.go

```diff
@@ -1703,14 +1703,24 @@ func addNewTasks(ctx context.Context, creationInfo TaskCreationInfo, existingBui
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
 		for _, taskName := range creationInfo.Pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if ok := existingTasksIndex[taskName]; ok {
+			if t, ok := existingTasksIndex[taskName]; ok {
+				if !t.Activated {
+					if err = SetActiveState(evergreen.DefaultTaskActivator, true, *t); err != nil {
```

Comment:
SetActiveState also updates build/version status each time it's called which seems a bit heavy. Can we instead save these taskIds / buildIds to include in the calls for updating build activation and task activation at the end of this function?

Reply:
That is fair - since `SetActiveState` accepts a list of tasks, I changed it to collect tasks we want to activate and then only invoke the function once after we finish collecting.